packages/framework/src/services/kafka/kafka-broker.client.test.ts

```diff
@@ -0,0 +1,89 @@
+import { Kafka } from 'kafkajs';
+import * as sinon from 'sinon';
+import { suite } from 'uvu';
+import * as assert from 'uvu/assert';
+
+import { KafkaBrokerClient, KafkaConsumePayload } from './kafka-broker.client';
+
+const test = suite('Kafka Broker Client');
+
+test.before.each(async (context) => {
+  context.sandbox = sinon.createSandbox();
+});
+
+test.after.each((context) => {
+  context.sandbox.restore();
+});
+
+test('should call connect', async (context) => {
+  const onProducerConnectSpy = context.sandbox.spy(context.sandbox.fake());
+  const onConsumerConnectSpy = context.sandbox.spy(context.sandbox.fake());
+
+  const kafkaProducer = context.sandbox.stub(Kafka.prototype, 'producer').returns({
+    connect: onProducerConnectSpy,
+  });
+  const kafkaConsumer = context.sandbox.stub(Kafka.prototype, 'consumer').returns({
+    connect: onConsumerConnectSpy,
+  });
+
+  const client = new KafkaBrokerClient('localhost:9092');
+  await client.connect();
+
+  assert.ok(kafkaProducer.calledOnce);
+  assert.ok(kafkaConsumer.calledOnce);
+  assert.ok(onProducerConnectSpy.calledOnce);
+  assert.ok(onConsumerConnectSpy.calledOnce);
+});
+
+test('should call subscribe', async (context) => {
+  const payload = {
+    topic: 'test_topic',
+    message: { value: new Buffer('test_value') },
+  };
+  const subscribeFunctionSpy = context.sandbox.spy((event: KafkaConsumePayload) => event);
+  const kafkaConsumerSubscribe = context.sandbox.spy(context.sandbox.fake());
+  const kafkaConsumerRun = context.sandbox.spy((config) => {
+    config.eachMessage(payload);
+  });
+
+  context.sandbox.stub(Kafka.prototype, 'producer').returns({
+    connect: context.sandbox.fake(),
+  });
+  context.sandbox.stub(Kafka.prototype, 'consumer').returns({
+    connect: context.sandbox.fake(),
+    subscribe: kafkaConsumerSubscribe,
+    run: kafkaConsumerRun,
+  });
+
+  const client = new KafkaBrokerClient('localhost:9092');
+  await client.connect();
+
+  await client.subscribe(subscribeFunctionSpy);
+
+  assert.ok(kafkaConsumerSubscribe.calledOnce);
+  assert.ok(kafkaConsumerRun.calledOnce);
+  assert.ok(subscribeFunctionSpy.calledOnce);
+  assert.ok(subscribeFunctionSpy.calledWith({ topic: payload.topic, message: payload.message.value }));
+});
+
+test('should call publish', async (context) => {
+  const kafkaProduceSend = context.sandbox.spy(context.sandbox.fake());
+
+  context.sandbox.stub(Kafka.prototype, 'producer').returns({
+    connect: context.sandbox.fake(),
+    send: kafkaProduceSend,
+  });
+  context.sandbox.stub(Kafka.prototype, 'consumer').returns({
+    connect: context.sandbox.fake(),
+  });
+
+  const client = new KafkaBrokerClient('localhost:9092');
+  await client.connect();
+
+  await client.publish({ topic: 'test_topic', messages: ['test_message'] });
```

Comment:
Would the kafka lib throw an error if we call `publish` without calling `connect` before?

Reply:
I have to try it. Not sure, but possible.

Reply:
I mean, of course, it will throw an error, just thinking it because the `connect` will set the producer and the consumer which won't be available when calling the publish or subscribe, but I think it's pretty much pointless to test it because the implementation is this way by spec.

Reply:
Cool